### PR TITLE
Add hyper_full example

### PR DESCRIPTION
**Pull request by Asakiz**
Signed-off-by: asakiz <asakizin@gmail.com>

### CODE REVIEW

File: hyper_full/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+#[async_trait::async_trait(?Send)]
+impl rwcst::LocalClientImpl for LocalClient {
+    type Err = Err;
+
+    fn new() -> Self {
+        LocalClient { client: hyper::Client::new() }
+    }
+
+    async fn fetch_info(&mut self) -> Result<rwcst::Info> {
+        let res = self.client.get("http://localhost:8001".parse().unwrap()).await?;
```
Comment (Jonathas-Conceicao) [inline hyper_full/src/main.rs]: Can this url parsing error be wrapped to the local error type?

File: hyper_full/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+#[async_trait::async_trait(?Send)]
+impl rwcst::RemoteClientImpl for RemoteClient {
+    type Err = Err;
+
+    fn new(remote: &str) -> Self {
+        RemoteClient { client: hyper::Client::new(), remote: remote.to_owned() }
+    }
+
+    async fn fetch_package(&mut self) -> Result<Option<(rwcst::Package, rwcst::Signature)>> {
+        let response = self.client.get("http://localhost:8001".parse().unwrap()).await?;
```
Comment (Jonathas-Conceicao) [inline hyper_full/src/main.rs]: Same with this parse error.

**Review by Jonathas-Conceicao — CHANGES_REQUESTED** [review]

File: hyper_full/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+[package]
+name = "hyper_full"
+version = "0.1.0"
+authors = ["asakiz <asakizin@gmail.com>"]
+edition = "2018"
+
+[dependencies]
+actix-rt = "1"
+actix-web = { version = "3.0.0-alpha.1", default-features = false }
```
Comment (otavio) [inline hyper_full/Cargo.toml]: Seems like there are unused dependencies on this. Take a look.

**Review by otavio — CHANGES_REQUESTED** [review]

File: hyper_full/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,141 @@
+// Copyright (C) 2020 O.S. Systems Sofware LTDA
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use derive_more::{Display, Error, From};
+use futures_util::lock::Mutex;
+use std::{convert::Infallible, sync::Arc};
+
+use bytes::buf::BufExt as _;
```
Comment (Jonathas-Conceicao) [inline hyper_full/src/main.rs]: Does this trait need to be hidden?

**Review by Jonathas-Conceicao — CHANGES_REQUESTED** [review]

File: hyper_full/src/main.rs (hunk 4 above)
Comment (Asakiz) [reply hyper_full/src/main.rs]: Actually is not.

**Review by Jonathas-Conceicao — APPROVED** [review]

**Review by otavio — APPROVED** [review]

**Pull request merged**